py/selenium/webdriver/firefox/service.py

```diff
@@ -43,12 +44,26 @@ def __init__(self, executable_path, port=0, service_args=None,
         """
         log_file = open(log_path, "a+") if log_path is not None and log_path != "" else None
 
+        if port != 0:
+            import warnings
+
+            warnings.warn(
+                "The 'port' parameter has been deprecated. Please pass in the port through "\
+                "service_args instead: ['--port', port]",
+                DeprecationWarning)
+        if service_args is not None and "--port" in service_args:
+            if port != 0:
+                raise ValueError("Deprecated parameter 'port' specified "\
+                    "simultaneously with '--port' argument in service_args")
+            else:
+                port = int(service_args[service_args.index("--port") + 1])
```

Comment:
hmm.. this is getting messy.. I hadn't realised that we're generating a port in the Python code based on the available ports, as geckodriver should accept 0 and bind that to an open port (though I'm not sure how we would determine the port it picks).

perhaps the best solution for now is to pass both `port` and `service_args` as you had, and raise an exception if `port` is not 0 and `--port` is in `service_args`.

I'd really like to hear @AutomatedTester's input on this though.